### PR TITLE
Add CORS support

### PR DESCRIPTION
**Pull request by chamil321**
This PR includes the CORS support in ballerina. The Resource modeling implementation is done according to the https://www.w3.org/TR/cors/. It supports both simple and preflight requests.

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/DispatcherUtil.java [hunk 1]
```diff
@@ -118,13 +122,17 @@ public static String getServiceBasePath(ServiceInfo service) {
...
         for (int x = 0; x < stringValues.size(); ++x) {
             builder.append(stringValues.get(x));
             if (x != stringValues.size() - 1) {
-                builder.append(", ");
+                if (spaceSeparated) {
+                    builder.append(" ");
+                } else {
+                    builder.append(", ");
+                }
             }
```
Comment (anupama-pathirage) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/DispatcherUtil.java]: Avoid this if else check within the loop. We can initialize the separator outside the loop based on the "spaceSeparated" value and then use it for the append

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/http/CorsHeaderGenerator.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        Map<String, List<String>> resourceCors;
+        if ((resourceCors = getResourceCors(cMsg, requestMethod)) == null) {
+            bLog.info(action + "headers are not declared properly");
+            return null;
+        }
+        if (!isEffectiveMethod(requestMethod, resourceCors.get(Constants.ALLOW_METHODS))) {
+            bLog.info(action + "not allowed method");
+            return null;
+        }
+
```
Comment (anupama-pathirage) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/http/CorsHeaderGenerator.java]: Remove extra new line here

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/DispatcherUtil.java (hunk 1 above)
Comment (chamil321) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/DispatcherUtil.java]: Thanks, Updated code

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/http/CorsHeaderGenerator.java (hunk 2 above)
Comment (chamil321) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/http/CorsHeaderGenerator.java]: Updated

**Pull request merged**